Assets/Scripts/Ability/AbilityManager.cs

```diff
@@ -242,6 +271,39 @@ internal void RemoveEffect()
             _slots[_slotNumber].transform.SetParent(null, true);
         }
 
+        private void StartCooldowns(float cooldown)
+        {
+            if (_abilityTokenSources[_slotNumber] != null)
+            {
+                _abilityTokenSources[_slotNumber]?.Cancel();
+                _abilityTokenSources[_slotNumber]?.Dispose();
+            }
+
+            _abilityTokenSources[_slotNumber] = new CancellationTokenSource();
```

Comment:
This really isn't how you're supposed to use this, as cancelling throws an exception.